[feature] Backport new msvc compiler settings to old build helpers and generators

<!-- What is your suggestion? Please be as specific as possible! -->

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

conan-center-index cannot ensure proper support of the new compiler `msvc` in `settings.yml` since all CCI recipes rely either on `cmake` + old `CMake` helper, or things like `AutoToolsBuildEnvironment`/`MSBuild`/`VisualStudioBuildEnvironment`/`Meson` helpers.

I've tested to support `libsodium` for example (https://github.com/conan-io/conan-center-index/pull/8957).

It builds with these 2 profiles:

```
[settings]
os=Windows
arch=x86_64
compiler=Visual Studio
compiler.version=17
build_type=Release
[options]
[build_requires]
[env]
```

```
[settings]
os=Windows
arch=x86_64
compiler=msvc
compiler.version=193
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.cppstd=17
build_type=Release
[env]
```

But the second one reports some linking incoherency in CMake based test_package (and not the first one):

```
cl : command line warning D9002: ignoring unknown option '/dynamic' [D:\projects\conan-center-index\recipes\libsodium\all\test_package\build\186009612552ee0133c675206a8ad5077496a98d\test_package.vcxproj]
  test_package.c
LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library [D:\projects\conan-center-index\recipes\libsodium\all\test_package\build\186009612552ee0133c675206a8ad5
077496a98d\test_package.vcxproj]
  test_package.vcxproj -> D:\projects\conan-center-index\recipes\libsodium\all\test_package\build\186009612552ee0133c675206a8ad5077496a98d\bin\test_package.exe
  Building Custom Rule D:/projects/conan-center-index/recipes/libsodium/all/test_package/CMakeLists.txt
```

So I guess there is some magic missing for this new compiler in old helpers or generators.

In this example I think there are 2 issues:
- something wrong at build time of the library which leads to the warning in test package: `LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library` (I guess that `MSBuild` has not properly honored `compiler.runtime`).
- and another one  in the old cmake generator or CMake helper in test package: `ignoring unknown option '/dynamic'`

--- a/conans/client/conf/detect.py
+++ b/conans/client/conf/detect.py
@@ -226,13 +226,6 @@ def _detect_compiler_version(result, output, profile_path):
         output.info("No compiler was detected (one may not be needed)")
         return
 
-    # Visual Studio 2022 onwards, detect as a new compiler "msvc"
-    if compiler == "Visual Studio":
-        version = Version(version)
-        if version == "17":
-            compiler = "msvc"
-            version = "193"
-
     result.append(("compiler", compiler))
     result.append(("compiler.version", _get_profile_compiler_version(compiler, version, output)))
 

--- a/conans/test/unittests/util/detect_test.py
+++ b/conans/test/unittests/util/detect_test.py
@@ -100,8 +100,5 @@ class DetectTest(unittest.TestCase):
             result = detect_defaults_settings(output=Mock(),
                                               profile_path=DEFAULT_PROFILE_NAME)
             result = dict(result)
-            self.assertEqual('msvc', result['compiler'])
-            self.assertEqual('193', result['compiler.version'])
-            self.assertEqual('14', result['compiler.cppstd'])
-            self.assertEqual('dynamic', result['compiler.runtime'])
-            self.assertEqual('Release', result['compiler.runtime_type'])
+            self.assertEqual('Visual Studio', result['compiler'])
+            self.assertEqual('17', result['compiler.version'])

